code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingUtils.java

```diff
@@ -106,6 +106,8 @@ static List<Object> toList(final JSONArray jsonArray) throws JSONException {
 
     /**
      * Converts provided {@link Object} to a {@link JSONObject} or {@link JSONArray}.
+     * This method is recursive.
```

Comment:
This method _may_ recurse.

Reply:
I'll fix this throughout the other recursive utility methods as well